apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/MonitoredResourceUtils.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+using Google.Api;
+using Google.Api.Gax;
+using System.Diagnostics;
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    internal static class MonitoredResourceUtils
+    {
+        /// <summary>The global resource.</summary>
+        public static readonly MonitoredResource GlobalResource = new MonitoredResource { Type = "global" };
+
+        /// <summary>
+        /// Attempts to detect the current <see cref="MonitoredResource"/> this is being run in.
+        /// If the monitored resource cannot be detected the global monitored resource is used.
+        /// </summary>
+        public static MonitoredResource DetectCurrentResource()
+        {
+            var platform = Platform.Instance();
```

Comment:
Just to make sure you're aware that Platform.Instance() can block for up to 1 second the first time it's called.